pkg/oauth/google/google.go

```diff
@@ -0,0 +1,70 @@
+package google
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/alexandrevilain/postgrest-auth/pkg/config"
+	"github.com/alexandrevilain/postgrest-auth/pkg/model"
+	"github.com/alexandrevilain/postgrest-auth/pkg/oauth"
+)
+
+type googleProvider struct {
+}
+
+//Googleuser struct of google user
+type Googleuser struct {
+	ID            string `json:"id"`
+	Email         string `json:"email"`
+	VerifiedEmail bool   `json:"verified_email"`
+	Name          string `json:"name"`
+	GivenName     string `json:"given_name"`
+	FamilyName    string `json:"family_name"`
+	Link          string `json:"link"`
+	Picture       string `json:"picture"`
+}
+
+// New retrive google user info based on the token provided
+func New() oauth.Provider {
+	return &googleProvider{}
+}
+
+// GetGoogleUserInfo retrive google user info based on the token provided
+func (provider *googleProvider) GetUserInfo(payload *oauth.Oauth2Payload, oauthStateString string, config *config.Config, db *sql.DB) (model.User, string, error) {
```

Comment:
Please, don't use the database here.
This method should only return a model.User.
Then, create the user in the HTTP handler. There is some duplicated code with Facebook.

The user creation shouldn't be done on a provider but must be generic.